### PR TITLE
Post/feed component

### PR DESCRIPTION
**Pull request by jakub-gosciniak**
# Related issue
- Post feed comonent to display user posts

# Changes
- Added FeedPost component
- Added moment package
- Added routing to user
- Added custom hook for content overflowing at window resizing
- Added avatar placeholder 
- Changed export default in firebase to export

# Additional information
- Add any additional information or context

# Checklist
- [x] Performed self-review
- [x] Added data-testId for UI elements
- [ ] Tests
- [x] Storybook

# Screenshots
image
image
image

### CODE REVIEW

File: src/components/FeedPost/FeedPost.scss [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+@import '../../styles/FontSizes';
+@import '../../styles/FontWeights';
+@import '../../styles/Breakpoints';
+
+.feedPost__container{
+    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
+    @include flexColumn;
+    max-inline-size: 500px;
+    border-radius: 10px;
+    background: white;
```
Comment (Polo11121) [inline src/components/FeedPost/FeedPost.scss:12]: all colors should be in Colors.scss !

File: src/components/FeedPost/FeedPost.scss [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    }
+}
+
+.feedPost__user{
+    display: flex;
+    align-items: center;
+    margin: 0;
+    width: 100%;
+    text-decoration: none;
+    color: black;
```
Comment (Polo11121) [inline src/components/FeedPost/FeedPost.scss:28]: Colors.scss : )

**Review by Polo11121 — DISMISSED** [review]
Looks clean, just remember that all colors should be in Colors.scss : )

File: src/styles/Colors.scss [hunk 3]
```diff
@@ -6,3 +6,5 @@ $gold: #e0b172;
 $lightGray: lightgray;
 $gray: gray;
 $mainTheme: linear-gradient(0deg, rgba(40, 103, 178, 0.28895308123249297) 0%, rgba(255, 255, 255, 1) 71%);
+$postBackgroundWhite: white;
+$postTextBlack:black;
```
Comment (MateuszLisowski) [inline src/styles/Colors.scss:10]:
```suggestion
$postTextBlack: black;
```

File: src/components/FeedPost/FeedPost.scss [hunk 4]
```diff
@@ -0,0 +1,83 @@
+@import '../../styles/Mixins';
+@import '../../styles/Colors';
+@import '../../styles/FontSizes';
+@import '../../styles/FontWeights';
+@import '../../styles/Breakpoints';
+
+.feedPost__container{
```
Comment (MateuszLisowski) [inline src/components/FeedPost/FeedPost.scss:7]: look at other files and components. There is always one root clas like "feedPost" and below there is somethig like this

```
.feedPost {
&__container {
...
}
&__user {
..
}
}
```

It should be definitely refactored

File: src/components/FeedPost/FeedPost.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  timestamp,
+  testid
+}: FeedPostProps) => {
+  const [seeMore, setSeeMore] = useState(false);
+  const ref = useRef() as MutableRefObject<HTMLParagraphElement>;
+  const isContentOverflowing = useIsContentOverflowing(ref);
+  const timePassed = moment(timestamp).startOf('hour').fromNow();
+
+  return (
+    <article className="feedPost__container">
```
Comment (mateusz0612) [inline src/components/FeedPost/FeedPost.tsx:31]: In my opinion it should be 'feedPost'

**Review by MatYouKy — CHANGES_REQUESTED** [review]
Please see the comments and make changes accordingly.

File: src/components/FeedPost/FeedPost.scss [hunk 6]
```diff
@@ -0,0 +1,85 @@
+@import '../../styles/Mixins';
+@import '../../styles/Colors';
+@import '../../styles/FontSizes';
+@import '../../styles/FontWeights';
+@import '../../styles/Breakpoints';
+
+.feedPost {
+  &__container {
```
[the remark this answers is not in the document]
Comment (MateuszLisowski) [reply src/components/FeedPost/FeedPost.scss:9]: I think it's not necessarry

File: src/styles/Colors.scss [hunk 7]
```diff
@@ -6,3 +6,5 @@ $gold: #e0b172;
 $lightGray: lightgray;
 $gray: gray;
 $mainTheme: linear-gradient(0deg, rgba(40, 103, 178, 0.28895308123249297) 0%, rgba(255, 255, 255, 1) 71%);
+$postBackgroundWhite: white;
+$postTextBlack: black;
```
Comment (MateuszLisowski) [inline src/styles/Colors.scss:10]:
```suggestion
$black: black;
```

File: src/styles/Colors.scss [hunk 8]
```diff
@@ -6,3 +6,5 @@ $gold: #e0b172;
 $lightGray: lightgray;
 $gray: gray;
 $mainTheme: linear-gradient(0deg, rgba(40, 103, 178, 0.28895308123249297) 0%, rgba(255, 255, 255, 1) 71%);
+$postBackgroundWhite: white;
```
Comment (MateuszLisowski) [inline src/styles/Colors.scss:9]:
```suggestion
$white: white;
```

File: src/stories/FeedPost/FeedPost.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    )
+  ]
+};
+
+const text1 =
+  'Jak to jest być skrybą, dobrze? A, wie pan, moim zdaniem to nie ma tak, że dobrze, albo że niedobrze. Gdybym miał powiedzieć, co cenię w życiu najbardziej, powiedziałbym, że ludzi. Ludzi, którzy podali mi pomocną dłoń, kiedy sobie nie radziłem, kiedy byłem sam, i co ciekawe, to właśnie przypadkowe spotkania wpływają na nasze życie. Chodzi o to, że kiedy wyznaje się pewne wartości, nawet pozornie uniwersalne, bywa, że nie znajduje się zrozumienia, które by tak rzec, które pomaga się nam rozwijać. Ja miałem szczęście, by tak rzec, ponieważ je znalazłem, i dziękuję życiu! Dziękuję mu; życie to śpiew, życie to taniec, życie to miłość! Wielu ludzi pyta mnie o to samo: ale jak ty to robisz, skąd czerpiesz tę radość? A ja odpowiadam, że to proste! To umiłowanie życia. To właśnie ono sprawia, że dzisiaj na przykład buduję maszyny, a jutro – kto wie? Dlaczego by nie – oddam się pracy społecznej i będę, ot, choćby, sadzić... doć— m-marchew...';
+const avatar =
+  'https://cdn.akamai.steamstatic.com/steamcommunity/public/images/avatars/bf/bfb953c0e4436d5400ea1d73d89c39c63a3285a1_full.jpg';
+const text = 'Social media bez memów? Ale jak to???';
```
Comment (MateuszLisowski) [inline src/stories/FeedPost/FeedPost.stories.tsx:21]: add enter for better readability

File: src/components/FeedPost/FeedPost.tsx (hunk 5 above)
Comment (MateuszLisowski) [reply src/components/FeedPost/FeedPost.tsx:31]: Agree, feedPost is missing here

File: src/components/FeedPost/FeedPost.scss [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+
+.feedPost {
+    
+  &__container {
+    @include flexColumn;
+    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans',
+      'Helvetica Neue', sans-serif;
+    max-inline-size: 500px;
+    border-radius: 10px;
+    background: white;
```
Comment (MateuszLisowski) [inline src/components/FeedPost/FeedPost.scss:15]: use $white

File: src/components/FeedPost/FeedPost.scss [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+@import '../../styles/Colors';
+@import '../../styles/FontSizes';
+@import '../../styles/FontWeights';
+@import '../../styles/Breakpoints';
+
+.feedPost {
+    
+  &__container {
+    @include flexColumn;
+    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans',
```
Comment (MateuszLisowski) [inline src/components/FeedPost/FeedPost.scss:11]: you could add it to Fonts

File: src/actions/useIsContentOverflowing.tsx [hunk 12]
```diff
@@ -0,0 +1,15 @@
+import { useState, useEffect, MutableRefObject } from 'react';
+
+export const useIsContentOverflowing = (ref: MutableRefObject<HTMLElement>) => {
+  const [isContentOverflowing, setIsContentOverflowing] = useState(false);
+
+  useEffect(() => {
+    function handleResize() {
```
Comment (MateuszLisowski) [inline src/actions/useIsContentOverflowing.tsx:7]: you could use arrow function here

File: src/components/FeedPost/FeedPost.scss [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+    width: 100%;
+    max-block-size: none;
+
+    &--see__less {
+      padding: 0;
+      max-block-size: 64px;
+      word-wrap: normal;
+      overflow: hidden;
+    }
+    @media only screen and (max-width: $mobile) {
```
[the remark this answers is not in the document]
Comment (MateuszLisowski) [reply src/components/FeedPost/FeedPost.scss:58]: I don't agree

**Review by MatYouKy — APPROVED** [review]

File: src/components/FeedPost/FeedPost.tsx [hunk 14]
```diff
@@ -0,0 +1,67 @@
+import React, { useRef, useState, MutableRefObject } from 'react';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import { useIsContentOverflowing } from '../../actions/useIsContentOverflowing';
+import './FeedPost.scss';
+
+interface FeedPostProps {
+  uid: string;
```
Comment (mateusz0612) [inline src/components/FeedPost/FeedPost.tsx:9]: I think it should be ownerUid like in firebase collection.

**Review by mateusz0612 — APPROVED** [review]
It looks good. Very nice that you added default image.

**Review by MatYouKy — APPROVED** [review]

File: src/styles/Fonts.scss [hunk 15]
```diff
@@ -0,0 +1,2 @@
+$postText:-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans',
```
Comment (MateuszLisowski) [inline src/styles/Fonts.scss:1]: I think we should reuse fonts so the name should be more general (not specifici lie postText

**Review by MateuszLisowski — APPROVED** [review]

**Review by Polo11121 — APPROVED** [review]
OK

**Pull request merged**